pennylane/gradients/gradient_transform.py

```diff
@@ -48,7 +48,10 @@ def gradient_expand(tape, depth=10):
             and ((supported_op(obj) and trainable_op(obj)) or not trainable_op(obj))
         )
 
-        return tape.expand(depth=depth, stop_at=stop_cond)
+        new_tape = tape.expand(depth=depth, stop_at=stop_cond)
+        params = new_tape.get_parameters(trainable_only=False)
+        new_tape.trainable_params = qml.math.get_trainable_indices(params)

```

Comment:
Turns out these two lines are required to solve a bug I discovered while writing the tests

Reply:
is this related to the `Unwrap` issue?

Reply:
I really hesitate to say this, but I often don't fully understand the trainable_params setting 😬 Each autodiff framework is different, it is affected by expansion, by higher order derivatives, etc. 

So a lot of the current logic is simply guided by 'this causes the tests to pass, for all interfaces, for all QNode variations, for all order derivatives, for all differentiation methods'.

Reply:
>So a lot of the current logic is simply guided by 'this causes the tests to pass, for all interfaces, for all QNode variations, for all order derivatives, for all differentiation methods'.

This is how I feel any time I have to write interface tests :sweat:

Reply:
Oh ok, could this be something to think about addressing for the upcoming planning? Or is it the case that since it only affects devs and we can generally muddle though that it's lower priority

Reply:
we all feel it 😿